### PR TITLE
tracee-ebpf: add support for unix socket in security_socket_* funcs

### PR DESCRIPTION
**Pull request by yanivagman**

### CODE REVIEW

**Review by grantseltzer — COMMENTED** [review]
Just a couple of questions on this, but it looks good and works how I expect.

Not indicative of a change in this PR, but in order to filter these events by AF_UNIX I had to do this:

`tracee-ebpf --trace security_socket_create.family=1 --trace event=security_socket_create`

Where '1' is AF_UNIX as defined in [socket.h](https://elixir.bootlin.com/linux/latest/source/include/linux/socket.h#L177). It'd be nice to be able to specify string values like 'AF_UNIX' instead.

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 1]
```diff
@@ -3066,6 +3076,16 @@ int BPF_KPROBE(trace_security_socket_accept)
 
         save_to_submit_buf(submit_p, (void *)&local, sizeof(struct sockaddr_in6), 1);
     }
+    else if (family == AF_UNIX) {
+        struct unix_sock *unix_sk = (struct unix_sock *)sk;
+        struct unix_address *addr = get_unix_sock_addr(unix_sk);
+        int len = READ_KERN(addr->len);
+        struct sockaddr_un sockaddr = {};
+        if (len <= sizeof(struct sockaddr_un)) {
```
[the remark this answers is not in the document]
Comment (yanivagman) [reply tracee-ebpf/tracee/tracee.bpf.c:3084]: Should never be false. It is here just to pass the verifier

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 2]
```diff
@@ -3066,6 +3076,16 @@ int BPF_KPROBE(trace_security_socket_accept)
 
         save_to_submit_buf(submit_p, (void *)&local, sizeof(struct sockaddr_in6), 1);
     }
+    else if (family == AF_UNIX) {
+        struct unix_sock *unix_sk = (struct unix_sock *)sk;
+        struct unix_address *addr = get_unix_sock_addr(unix_sk);
+        int len = READ_KERN(addr->len);
```
[the remark this answers is not in the document]
Comment (yanivagman) [reply tracee-ebpf/tracee/tracee.bpf.c:3082]: No reason. I'll do that

**Comment by yanivagman** [comment]
> It'd be nice to be able to specify string values like 'AF_UNIX' instead.

Good idea. We can think about doing this in a future PR (it is somehow the reverse of `prepareArgs()`)

**Comment by grantseltzer** [comment]
> > It'd be nice to be able to specify string values like 'AF_UNIX' instead.
> 
> Good idea. We can think about doing this in a future PR (it is somehow the reverse of `prepareArgs()`)

I'll make an issue!

**Review by grantseltzer — APPROVED** [review]
LGTM

**Pull request merged**